database/migrations/000032_provider_class.up.sql

```diff
@@ -0,0 +1,35 @@
+-- Copyright 2024 Stacklok, Inc
+--
+-- Licensed under the Apache License, Version 2.0 (the "License");
+-- you may not use this file except in compliance with the License.
+-- You may obtain a copy of the License at
+--
+--      http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+
+CREATE TYPE provider_class AS enum ('github', 'gh-app');
+
+ALTER TABLE providers
+  -- Fills existing rows with 'github'
+  ADD COLUMN class provider_class DEFAULT 'github' NOT NULL;
+
+-- providers becomes a star schema, where the provider is the central table,
+-- and different classes of providers have different provider_* tables that
+-- join with the providers table to get the credential parameters.
+
+CREATE TABLE provider_app_installations (
+    app_installation_id TEXT PRIMARY KEY,
+    provider_id UUID,
+    organization_id BIGINT NOT NULL,
```

Comment:
Is this table mostly just meant for GitHub? This organization ID seems quite github specific.

Reply:
I'd actually prefer to have more tables and separate the implementations for each provider more, potentially to different storage for different providers in the future (to enable commercial or off-cluster providers).